packages/client/lib/sync/fullsync.ts

```diff
@@ -95,69 +92,62 @@ export class FullSynchronizer extends Synchronizer {
    * @return Resolves when sync completed
    */
   async syncWithPeer(peer?: Peer): Promise<boolean> {
-    if (!peer) return false
-    const latest = await this.latest(peer)
-    if (!latest) return false
-    const height = new BN(latest.number)
-    const first = this.chain.blocks.height.addn(1)
-    const count = height.sub(first).addn(1)
-    if (count.lten(0)) return false
-
-    this.config.logger.debug(
-      `Syncing with peer: ${peer.toString(true)} height=${height.toString(10)}`
-    )
-
-    this.blockFetcher = new BlockFetcher({
-      config: this.config,
-      pool: this.pool,
-      chain: this.chain,
-      interval: this.interval,
-      first,
-      count,
-    })
-    this.config.events.on(Event.SYNC_FETCHER_FETCHED, (blocks) => {
-      blocks = blocks as Block[]
-      const first = new BN(blocks[0].header.number)
-      const hash = short(blocks[0].hash())
-      const baseFeeAdd = this.config.chainCommon.gteHardfork('london')
-        ? `basefee=${blocks[0].header.baseFeePerGas} `
-        : ''
-      this.config.logger.info(
-        `Imported blocks count=${blocks.length} number=${first.toString(
-          10
-        )} hash=${hash} ${baseFeeAdd}hardfork=${this.config.chainCommon.hardfork()} peers=${
-          this.pool.size
-        }`
+    // eslint-disable-next-line no-async-promise-executor
+    return await new Promise(async () => {
```

Comment:
hm there is no resolve or reject on this promise

Reply:
hm, maybe this is a place where `Event.SYNC_SYNCHRONIZED` would be useful, we could resolve when received if the typedoc above is still accurate: `* @return Resolves when sync completed`